tests/flytekit/common/workflows/sagemaker.py

```diff
@@ -0,0 +1,122 @@
+)
+from flytekit.models.sagemaker.parameter_ranges import (
+    ContinuousParameterRange,
+    HyperparameterScalingType,
+    IntegerParameterRange,
+    ParameterRangeOneOf,
+)
+from flytekit.models.sagemaker.training_job import (
+    AlgorithmName,
+    AlgorithmSpecification,
+    InputContentType,
+    InputMode,
+    TrainingJobResourceConfig,
+)
+from flytekit.sdk.types import Types
+from flytekit.sdk.workflow import Input, workflow_class
+
+example_hyperparams = {
+    "base_score": "0.5",
+    "booster": "gbtree",
+    "csv_weights": "0",
+    "dsplit": "row",
+    "grow_policy": "depthwise",
+    "lambda_bias": "0.0",
+    "max_bin": "256",
+    "max_leaves": "0",
+    "normalize_type": "tree",
+    "objective": "reg:linear",
+    "one_drop": "0",
+    "prob_buffer_row": "1.0",
+    "process_type": "default",
+    "rate_drop": "0.0",
+    "refresh_leaf": "1",
+    "sample_type": "uniform",
+    "scale_pos_weight": "1.0",
+    "silent": "0",
+    "sketch_eps": "0.03",
+    "skip_drop": "0.0",
+    "tree_method": "auto",
+    "tweedie_variance_power": "1.5",
+    "updater": "grow_colmaker,prune",
+}
+
+builtin_algorithm_training_job_task2 = SdkBuiltinAlgorithmTrainingJobTask(
+    training_job_resource_config=TrainingJobResourceConfig(
+        instance_type="ml.m4.xlarge", instance_count=1, volume_size_in_gb=25,
+    ),
+    algorithm_specification=AlgorithmSpecification(
+        input_mode=InputMode.FILE,
+        input_content_type=InputContentType.TEXT_CSV,
+        algorithm_name=AlgorithmName.XGBOOST,
+        algorithm_version="0.72",
+    ),
+)
+
+simple_xgboost_hpo_job_task = hpo_job_task.SdkSimpleHyperparameterTuningJobTask(
+    training_job=builtin_algorithm_training_job_task2,
+    max_number_of_training_jobs=10,
+    max_parallel_training_jobs=5,
+    cache_version="1",
+    retries=2,
+    cacheable=True,
+    tunable_parameters=["num_round", "max_depth", "gamma"],
+)
+
+
+@workflow_class
+class SageMakerHPO(object):
+    train_dataset = Input(Types.MultiPartCSV, default="s3://somelocation")
+    validation_dataset = Input(Types.MultiPartCSV, default="s3://somelocation")
+    static_hyperparameters = Input(Types.Generic, default=example_hyperparams)
+    hyperparameter_tuning_job_config = Input(
+        HyperparameterTuningJobConfig,
+        default=_HyperparameterTuningJobConfig(
+            tuning_strategy=HyperparameterTuningStrategy.BAYESIAN,
+            tuning_objective=HyperparameterTuningObjective(
+                objective_type=HyperparameterTuningObjectiveType.MINIMIZE, metric_name="validation:error",
+            ),
+            training_job_early_stopping_type=TrainingJobEarlyStoppingType.AUTO,
+        ),
+    )
+
+    a = simple_xgboost_hpo_job_task(
+        train=train_dataset,
+        validation=validation_dataset,
+        static_hyperparameters=static_hyperparameters,
+        hyperparameter_tuning_job_config=hyperparameter_tuning_job_config,
+        num_round=ParameterRangeOneOf(
```

Comment:
Sorry I don't think I understand the code so these might just be stupid questions. But when should the user use ParameterRangeOneOf and when should one use ParameterRange?

Reply:
ParameterRange is the type... you use it when you declare the interface of a task/WF... etc...
ParameterRangeOneOf is the value you can set for that type.... 

We can choose to name them the same but keep them in two different packages... (the flytekit way as far as I understand)...

@wild-endeavor  might help shed some light on how to properly do this...